scripts/generator/graphql_generator/csharp/SDK/ErrorFormatter.cs.erb

```diff
@@ -0,0 +1,22 @@
+namespace <%= namespace %> {
+    using <%= namespace %>.SDK;
+
+    public static class ErrorFormatter {
+        /// <summary>
+        /// Computes the ShopifyError we want to give back to the developer from a response. This method will return the first available error.
+        /// </summary>
+        /// <param name="response">response to format an error from</param>
+        public static ShopifyError FormatErrorFromResponse(TopLevelResponse response) {
+            if (response.HTTPError != null) {
+                return new ShopifyError(ShopifyError.ErrorType.HTTP, response.HTTPError);
+            } 
+            
+            if (response.errors != null && response.errors.Count > 0) {
+                // Take the first error we received from GraphQL and bubble it up.
+                return new ShopifyError(ShopifyError.ErrorType.GraphQL, response.errors[0]);
```

Comment:
We can't just pass the first error back. There maybe multiple syntax errors from graphql. I think at a minimum we should just concat all errors with newlines.

I need to think about this more.